### PR TITLE
Update procedures documentation

### PR DESCRIPTION
**Pull request by pontusmelke**

### CODE REVIEW

File: manual/embedded-examples/src/docs/dev/procedures.asciidoc [hunk 1]
```diff
@@ -221,30 +222,47 @@ package example;
...
+
+            // When I use the index procedure to index a node
+            session.run( "CALL example.index({id}, ['name'])", parameters( "id", nodeId ) );
+
+            // Then I can search for that node with lucene query syntax
+            StatementResult result = session.run( "CALL example.search('User', 'name:Brook*')" );
+            assertThat( result.single().get( "nodeId" ).asLong(), equalTo( nodeId ) );
+        }
+    }
 }
```
Comment (chrisvest) [inline manual/embedded-examples/src/docs/dev/procedures.asciidoc:266]: Have you thought about turning these code examples into snippets? As in, put them in actual Java files somewhere, so we can make sure that they always compile.

Comment (jjaderberg) [inline manual/embedded-examples/src/docs/dev/procedures.asciidoc:266]: We definitely want to add verification of some kind. The source lives in a separate repository, however, so it is non-trivial. We have a card for adding this later.

File: manual/embedded-examples/src/docs/dev/procedures.asciidoc [hunk 2]
```diff
@@ -46,15 +46,16 @@ These are listed in the table below:
 
 [options="header", cols="m,d"]
```
Comment (jjaderberg) [inline manual/embedded-examples/src/docs/dev/procedures.asciidoc]: This line is wrong, `cols="m,d"` means there are only two columns in the table, but there are three. A recent commit removed one of the column specifiers, adding it back should fix the table (i.e., making it `cols="m,m,d"`).

**Comment by jjaderberg** [comment]
Not part of the original scope of this PR, but it would be lovely if you could change "legacy index" to "manual index" (two occurrences), as that's what we will call those indexes going forward.

File: manual/embedded-examples/src/docs/dev/procedures.asciidoc (hunk 1 above)
Comment (pontusmelke) [inline manual/embedded-examples/src/docs/dev/procedures.asciidoc:266]: @chrisvest We have a card for doing that, we will get to that directly after release

**Comment by pontusmelke** [comment]
@jjaderberg Fixed your comments

**Comment by jjaderberg** [comment]
Thanks! The table now renders correctly, this is ready to be merged as far as I'm concerned.

**Pull request merged**